app/helpers/holdings_helper.rb

```diff
@@ -18,11 +18,18 @@ def holdings(document, method)
   end
 
   def location_name(item)
-    @locations.dig(item[:location].downcase.gsub('_', ''), 'name') || 'Unknown'
+    Location.find_by!(short_code: item[:location].downcase.delete('_').to_sym).name
+  rescue ActiveRecord::RecordNotFound => e
+    Rollbar.error("Error retriving name for Location #{item[:location].downcase.delete('_').to_sym}", e)
+    Location.create(
+      short_code: item[:location].downcase.delete('_').to_sym,
+      name: 'Unknown'
+    )
+    'Unknown'
   end
 
   def location_url(item)
-    @locations.dig(item[:location].downcase.gsub('_', ''), 'locationurl')
+    Location.find_by(short_code: item[:location].downcase.gsub('_', ''))&.url
```

Comment:
What do we want to happen if the item[:location] can't be found?

Reply:
Every item *should* have a location - they're catalogued with a location, so that's a fair assumption for us to make. If a location can't be found, ideally this would fail gracefully and an error reported to us somehow. We've never done content-based error reporting, however, so this might be a new thing for us. At minimum it should fail gracefully, in which case a NEOS library will report the error to us or fix it at the cataloguing level.